custom_components/solcast_solar/coordinator.py

```diff
@@ -95,27 +101,25 @@ async def update_utcmidnight_usage_sensor_data(self, *args) -> None:
             #_LOGGER.error("Exception in update_utcmidnight_usage_sensor_data(): %s", traceback.format_exc())
             pass
 
-    async def update_midnight_spline_recalc(self, *args) -> None:
+    async def update_midnight_spline_recalc(self, *args):
         """Re-calculates splines at midnight local time."""
         try:
-            _LOGGER.debug("Recalculating splines")
-            await self.solcast.spline_moments()
-            await self.solcast.spline_remaining()
+            await self.solcast.recalculate_splines()
         except:
             _LOGGER.error("Exception in update_midnight_spline_recalc(): %s", traceback.format_exc())
 
-    async def service_event_update(self, *args) -> None:
+    async def service_event_update(self, *args):
         """Get updated forecast data when requested by a service call."""
         try:
             #await self.solcast.sites_weather()
-            await self.solcast.http_data(dopast=False)
```

Comment:
I assume that this is a correct replacement of using http_data with get_forecast_update - but it stood out so I figured I'd check rather than not ask the "dumb" question.

Reply:
Replaced with this, so yes it has been renamed.

```
await self.solcast.get_forecast_update(do_past=False)
```

I have stuffed something up with sensors, so don't run this code just yet.